packages/lambda-powertools-middleware-correlation-ids/event-sources/api-gateway.js

```diff
@@ -3,7 +3,7 @@ const Log = require('@dazn/lambda-powertools-logger')
 const consts = require('../consts')
 
 function isMatch (event) {
-  return event.hasOwnProperty('httpMethod')
+  return event.hasOwnProperty('httpMethod') && !event.requestContext.hasOwnProperty('elb')
```

Comment:
we won't need this if we move the alb event source before API GW in the `eventSources` array in `index.js`

Reply:
mm.. I guess, the benefit of having it is so that we're not prone to problems with ordering in an array.

Reply:
Yes exactly, that was the thought behind this - for this bit to detect API GW on it's own.

Reply:
the downside here is that the API Gateway event source thingie needs to check for ALB event source thingie, which is a bit odd

Reply:
The events are very similar to each other so I think this should be ok to ensure we aren't mistakingly identifying the wrong event. Any suggestions to  improve this?

Reply:
not really, let's leave it as it is